src/config/file/v1.rs

```diff
@@ -190,10 +226,41 @@ impl TryFrom<Integration> for remote::Integration {
                     .team(team)
                     .title(title)
                     .protocol(protocol)
+                    .import_branches(import_branches)
+                    .import_tags(import_tags)
+                    .watched_branches(watched_branches)
                     .build()
-                    .wrap_ok()
+            }
+        };
+
+        if integration.watched_branches().is_empty()
+            && integration.import_branches() == &remote::BranchImportStrategy::Enabled
+        {
+            let references =
+                futures::executor::block_on(integration.references()).unwrap_or_default();
+            let primary_branch = references
+                .iter()
+                .find(|r| r.name() == MAIN_BRANCH || r.name() == MASTER_BRANCH)
+                .cloned();
+
+            match primary_branch {
+                None => {
+                    // Watched branches was not set and failed to infer a primary branch
+                    return report!(remote::ValidationError::PrimaryBranch)
+                        .wrap_err()
+                        .help("Consider providing explicit values for watched branches in the integration config")
+                        .describe_lazy(|| "infer watched branches")
+                        .documentation_lazy(doc::link::config_file_reference)?;
+                }
+                Some(branch) => {
+                    let primary_branch_name = branch.name();
+                    warn!("Inferred '{primary_branch_name}' as the primary branch for '{integration:#?}'. Broker imports only the primary branch when inferred; if desired this can be customized with the integration configuration.");
```

Comment:
In most of the rest of the code we use the `Display` trait for `Integration` when logging it (this line uses the `Debug` trait). Why are we switching it up here?